exercises/pythagorean-triplet/src/test/java/PythagoreanTripletTest.java

```diff
@@ -0,0 +1,95 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import org.junit.Ignore;
+
+public class PythagoreanTripletTest {
+
+    @Test
+    public void shouldCalculateSum() {
+        PythagoreanTriplet sut = new PythagoreanTriplet(3, 4, 5);
+        final int expected = 12;
+        final int actual = sut.calculateSum();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldCalculateProduct() {
+        PythagoreanTriplet sut = new PythagoreanTriplet(3, 4, 5);
+        final long expected = 60l;
+        final long actual = sut.calculateProduct();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void testIsPythagoreanOK() {
+        PythagoreanTriplet sut = new PythagoreanTriplet(3, 4, 5);
+        assertTrue(sut.isPythagorean());
+    }
+
+    @Test
+    @Ignore
+    public void testIsPythagoreanFail() {
+        PythagoreanTriplet sut = new PythagoreanTriplet(5, 6, 7);
+        assertFalse(sut.isPythagorean());
+    }
+
+    @Test
+    @Ignore
+    public void shouldMakeTripletsUpToTen() {
+        final Map<String, Integer> searchData
+                = new HashMap<>();
+        searchData.put(PythagoreanTriplet.MAX_FACTOR, 10);
+        final List<Long> actual
+                = PythagoreanTriplet.createTripletsList(searchData)
+                        .stream()
+                        .map(t -> t.calculateProduct())
+                        .sorted((p1, p2) -> Double.compare(p1, p2))
```

Comment:
If elements implement Comparable, they are sorted in natural order, and the ClassCastException may be thrown otherwise. In our case we have Long elements, so can get by without the comparator. :)